qemu/tests/block_kill_reconnect_with_remote_storage.py

```diff
@@ -0,0 +1,157 @@
+        if result.exit_status != 0:
+            test.error('command error: %s' % result.stderr.decode())
+
+    def run_iptables(cmd):
+        result = process.run(cmd, ignore_status=True, shell=True)
+        if result.exit_status != 0:
+            logging.error('command error: %s' % result.stderr.decode())
+
+    def break_net_with_iptables():
+        run_iptables(params['net_break_cmd'])
+        net_down = True
+
+    def resume_net_with_iptables():
+        run_iptables(params['net_resume_cmd'])
+        net_down = False
+
+    def reconnect_loop_io(test_disk):
+        error_context.context(
+            "Run IO test when in reconnecting loop", logging.info)
+        for iteration in range(repeat_times):
+            error_context.context("Wait %s seconds" % reconnect_time_wait,
+                                  logging.info)
+            time.sleep(reconnect_time_wait)
+            run_io_test(test_disk)
+
+    def export_data_img():
+        logging.info("Resume to export nbd data disks.")
+        logging.info("Run command: %s" % resume_export_img_cmd)
+        result = os.system(resume_export_img_cmd)
+        if result != 0:
+            test.error('command error: %s' % result.stderr.decode())
+        logging.info("Wait %s seconds" % reconnect_time_wait)
+        time.sleep(reconnect_time_wait)
+
+    def check_data_disk_resume(test_disk):
+        error_context.context(
+            "check data disk resumed", logging.info)
+        for iteration in range(repeat_times):
+            logging.info("Wait %s seconds" % reconnect_time_wait)
+            time.sleep(reconnect_time_wait)
+            run_io_test(test_disk)
+
+    def clean_images(net_status):
+        """
+        recover nbd image access
+        """
+        if net_status:
+            resume_net_with_iptables()
+
+    disk_op_cmd = params.get("disk_op_cmd")
+    disk_op_timeout = int(params.get("disk_op_timeout", 360))
+    timeout = int(params.get("login_timeout", 360))
+    save_export_data_img_cmd = params.get("save_export_data_img_cmd")
+    find_data_disk_pid_cmd = params.get("find_data_disk_pid_cmd")
+    find_data_disk_cmd = params.get("find_data_disk_cmd")
+    find_system_disk_cmd = params.get("find_system_disk_cmd")
+    nbd_port_data1 = params.get("nbd_port_data1")
+    repeat_times = int(params.get("repeat_times"))
+    reconnect_time_wait = int(params.get("reconnect_time_wait"))
+    error_context.context(find_system_disk_cmd, logging.info)
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+
+    disk_storage_name = get_disk_storage_name(find_system_disk_cmd,
+                                              find_data_disk_cmd)
+    system_disk = disk_storage_name[0]
+    data_disk = disk_storage_name[1]
+    if disk_op_cmd:
+        run_io_test(data_disk)
+    resume_export_img_cmd = save_data_img_process(nbd_port_data1)
+    net_down = False
+    kill_data_disk(nbd_port_data1)
+    break_net_with_iptables()
```

Comment:
why do we need to break net here? For data1 has already offlined by killing qemu-nbd(nbd export stopped)

Reply:
what about breaking net only?i.e. don't kill qemu-nbd, in this situation, we also lose connection to data1

Reply:
Kill qemu-nbd is a must for this test, as recent qemu 5.2 code changes about reconnect-delay fix an issue of qemu hang when socket dead within reconnecting loop, refer to block/nbd: fix reconnect-delay.